src/error.rs

```diff
@@ -166,3 +169,196 @@ impl From<BitcoinEncodeError> for EncodeError {
         }
     }
 }
+
+#[derive(Debug, thiserror::Error, uniffi::Error)]
+pub enum PsbtError {
```

Comment:
I believe remote error types can be used with `uniffi::remote(Enum)`. Then I'd guess they don't need to derive `thiserror::Error`, or require a conversion, or require the explicit display #[error] attributes. Not sure If it has been tried in this repo before. I think an enum still needs to be written with each variant but some of the other boilerplate that might get out of sync can go. I am not certain of this but the documentation suggests it.

https://mozilla.github.io/uniffi-rs/latest/types/remote_ext_types.html

Reply:
With the Psbt Error Enum there are a lot of complex error payload types, wouldn't these all still need to be converted for this to be easily used as a `remote(Enum)`? I'm just not sure its worth it to avoid error messages getting out of sync unless there is a way to more easily avoid those conversions.

Reply:
I think [`#[uniffi(flat_error)]`](https://mozilla.github.io/uniffi-rs/latest/proc_macro/errors.html) might be able to keep the upstream payload types but inherit debug/display implementations (perhaps with `#[uniffi::export(Debug), uniffi::export(Display)]`, not sure to what extent these derives can be combined). You may need to rely on the `bitcoin/std` feature so errors are `std::error::Error` in order to make sure UniFFI knows the remote `PsbtError` can be thrown as an error.

Reply:
Errors have been a pain in our side for a long time, but I'm curious to see if some of the new features in uniffi might fix some of those pain points for us. In particular, simply using the remote error types would be awesome instead of reimplementing errors.

For a comprehensive dive into this from a year ago (some things have changed since then so I still have hope!) see this issue: https://github.com/bitcoindevkit/bdk-ffi/issues/509.

Reply:
From what I remember our problem was mostly about having to choose between offering the complex types in the payload vs correct and interesting error messages for users (which they expect in their target languages).

Reply:
Note that you cannot use `#[uniffi::export(Display)]` on enums (you get the error `unsupported item: only functions and impl blocks may be annotated with this attribute`).

Reply:
Ok weird. Maybe I got this wrong. I tried a bunch of stuff 2 weeks ago to play with the errors a bit more and got to the same place we ended up so far. 

Let me know how you make out I'm definitely interested in better errors!